tests/staticfiles_tests/test_storage.py

```diff
@@ -463,6 +464,48 @@ def test_hashed_name(self):
         self.assertEqual(relpath, 'cached/styles.css')
 
 
+class TestCustomManifestStorage(SimpleTestCase):
+    def setUp(self):
+        staticfiles_storage_class = storage.ManifestStaticFilesStorage
+
+        self.manifest_path = Path(tempfile.mkdtemp())
+        self.manifest_file = self.manifest_path / staticfiles_storage_class.manifest_name
+        self.manifest = {
+            'version': staticfiles_storage_class.manifest_version,
+        }
+        with self.manifest_file.open('w') as manifest_file:
+            json.dump(self.manifest, manifest_file)
+
+        self.staticfiles_storage = staticfiles_storage_class(
+            manifest_storage=staticfiles_storage_class(location=self.manifest_path)
+        )
+
+    def tearDown(self):
+        try:
+            shutil.rmtree(self.manifest_path)
+        except FileNotFoundError:
+            self.assertEqual(self.removed_manifest, True)
+
+    def test_read_manifest_existing(self):
+        self.assertEqual(self.staticfiles_storage.read_manifest(), json.dumps(self.manifest))
+
+    def test_read_manifest_missing(self):
+        os.remove(self.manifest_file)
+        self.removed_manifest = True
+        self.assertIsNone(self.staticfiles_storage.read_manifest())
+
+    def test_save_manifest_override(self):
+        self.staticfiles_storage.save_manifest()
+        with open(self.manifest_file) as manifest_file:
+            manifest = json.loads(manifest_file.read())
+        self.assertLessEqual(self.manifest.items(), manifest.items())
+
+    def test_save_manifest_create(self):
+        os.remove(self.manifest_file)
+        self.staticfiles_storage.save_manifest()
+        self.assertTrue(os.path.isfile(self.manifest_file))
```

Comment:
Django style is to use assertEqual to check booleans to ensure truthy expressions don't slip by, although here the expression is coming straight from os.path, so bit of a nit pick but still worth doing IMO.

Reply:
I was commenting off the top of my head and got the style guide wrong, it actually suggests `assertIs()` now that I look at it, which makes sense. Sorry! Don't feel obligated to tinker with this again on my account; feel free to wait for a substantive final review.